src/averageAges.js

```diff
@@ -36,8 +41,17 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filteredPeople = withChildren
+    ? people.filter(({ sex, name }) => (
+      people.some(({ mother }) => mother === name)) && sex === 'f')
+    : people.filter(({ sex }) => sex === 'f');
+  const length = filteredPeople.length;
+
+  return ((filteredPeople
+    .reduce((sumAges, { born, died }) => sumAges + (died - born), 0)) / length
+  );
```

Comment:
I thought about creating two separate variables. I agree that the code would be more clear. But in this case, we always have to filter the array twice to create variables. Regardless of `withChildren` parameter. Or use `if` again))

Reply:
As I see now you also use `filter` twice. The difference only, that you initialise one variable instead of 2. But now code looks better(common vision) 👍